tracer/src/Datadog.Trace/Logging/Internal/DatadogSerilogLogger.cs

```diff
@@ -20,13 +20,16 @@ internal class DatadogSerilogLogger : IDatadogLogger
         private readonly ILogRateLimiter _rateLimiter;
         private ILogger _logger;
 
-        public DatadogSerilogLogger(ILogger logger, ILogRateLimiter rateLimiter)
+        public DatadogSerilogLogger(ILogger logger, ILogRateLimiter rateLimiter, string fileLogDirectory)
         {
             _logger = logger;
             _rateLimiter = rateLimiter;
+            FileLogDirectory = fileLogDirectory;
         }
 
-        public static DatadogSerilogLogger NullLogger { get; } = new(SilentLogger.Instance, new NullLogRateLimiter());
+        public static DatadogSerilogLogger NullLogger { get; } = new(SilentLogger.Instance, new NullLogRateLimiter(), null);
+
+        public string FileLogDirectory { get; }
```

Comment:
Having nullable enabled in some parts and not others is a bit scary. Here, FileLogDirectory can be null but we're losing this information :(

Reply:
Yeah... I can't wait for us to nullable enable everything at this point... I might do a separate PR to add nullable support to all the logging stuff to avoid noise (and then fix this)

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/4994 - I'll stack the other PRs onto it